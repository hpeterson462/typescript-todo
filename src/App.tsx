import React, { useState } from 'react';
import './App.css';
import { v4 as uuid } from 'uuid';
import Header from './components/Header';
import TodoInput from './components/TodoInput';
import TodoList from './components/TodoList';
import CompletedList from './components/CompletedList';

export interface ITodo {
  id: any;
  description: string;
  isDone: boolean;
}

export class Todo implements ITodo {
  id: any;
  description: string;
  isDone: boolean;

  constructor(description: string = '', isDone: boolean = false) {
    this.id = uuid();
    this.description = description;
    this.isDone = isDone;
  }
}

function App() {

  const newTodos = [
    new Todo('')
  ];

  const [todos, setTodos] = useState(newTodos);

  function addTodo(todo: Todo) {
    setTodos([todo, ...todos]);
  }

  return (
    <div>
      <Header />
      <TodoInput addTodo={addTodo} />
      <TodoList todos={todos} />
      <CompletedList todos={todos} />
    </div>
  );
}

export default App;
