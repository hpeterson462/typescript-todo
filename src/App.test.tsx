import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('App component', () => {
  it('renders Todo List', () => {
    render(<App />);
    const todoElement = screen.getByText(/todo list/i);
    expect(todoElement).toBeInTheDocument();
  });

  it('renders App', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  })
});
