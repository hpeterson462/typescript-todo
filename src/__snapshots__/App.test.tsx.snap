// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div>
    <header>
      <h1>
        Todo List
      </h1>
    </header>
    <form
      id="todo-input"
    >
      <input
        name="todo"
        placeholder="Add new item"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add
      </button>
    </form>
    <div>
      <h2>
        Todo
      </h2>
      <ul>
        <li>
          <input
            type="checkbox"
          />
        </li>
      </ul>
    </div>
    <div>
      <h2>
        Completed!
      </h2>
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
